File: RemoveWord/Ruby/RemoveWord.rb
class RemoveWord
    def remove_word_from_sentence(sentence, word)
        def remove_word(sentence, word)
            # return nothing if no sentence
            return sentence if sentence.empty?
          
            # remove all multiple spaces
            sentence = sentence.split.join(' ')
          
            # return sentence to word is empty
            return sentence if word.empty?
          
            # remove punctuation
            sentence = sentence.sub(/[^\w\s]/, '')

            # case insensitive replacement
            word_list = sentence.split(' ')
            sentence = word_list.reject! { |x| word.downcase == x.downcase }.join(' ')

            # remove all multiple spaces
            sentence = sentence.split.join(' ')

            # strip leading and trailing space
            sentence = sentence.strip
          
            sentence
          end
    end
end

File: RemoveWord/Python/removeWord.py
import sys
import re

def remove_word_from_sentence(sentence, word):
    # return nothing if no sentence
    if not sentence:
        return sentence
        
    sentence = remove_multiple_spaces(sentence)

    # return sentence to word is empty
    if not word: 
        return sentence
    
    # remove punctuation
    sentence = re.sub(r'[^\w\s]', '', sentence)
    
    # case insensitive replacement
    word_list = sentence.split()
    new_words = list(filter(lambda w: w.lower() != word.lower(), word_list))
    sentence =" ".join(new_words)

    sentence = remove_multiple_spaces(sentence)

    return sentence.strip()

def remove_multiple_spaces(sentence):
    # remove all multiple spaces
    while sentence.find("  ") > -1:
        sentence = sentence.replace("  ", " ")
    return sentence